packages/engine/src/behave-graph/nodes/Profiles/Engine/helper/actionHelper.ts

```diff
@@ -105,3 +119,94 @@ export function getActionDispatchers() {
   }
   return dispatchers
 }
+
+type State = {
+  queue: ActionQueueDefinition
+  systemUUID: SystemUUID
+}
+const initialState = (): State => ({
+  queue: undefined!,
+  systemUUID: '' as SystemUUID
+})
+let systemCounter = 0
+export function getActionConsumers() {
+  const consumers: NodeDefinition[] = []
+  const skipped: string[] = []
+  for (const [actionType, action] of Object.entries(ActionDefinitions)) {
+    const { type, ...actionDef } = action.actionShape
+    if (skipAction.includes(actionType)) {
+      skipped.push(actionType)
+      continue
+    }
+    const outputSockets = generateActionNodeSchema(actionDef)
+    if (Object.keys(outputSockets).length === 0) {
+      skipped.push(actionType)
+      continue
+    }
+    const nameArray = type.split('.')
+    const dispatchName = startCase(nameArray.pop().toLowerCase())
+    const namePath = nameArray.splice(1).join('/')
+
+    const node = makeEventNodeDefinition({
+      typeName: `action/${namePath}/on${dispatchName}`,
+      category: NodeCategory.Event,
+      label: `on ${namePath} ${dispatchName}`,
+      in: {
+        system: (_, graph) => {
+          const systemDefinitions = Array.from(SystemDefinitions.keys()).map((key) => key as string)
+          const groups = systemDefinitions.filter((key) => key.includes('group')).sort()
+          const nonGroups = systemDefinitions.filter((key) => !key.includes('group')).sort()
+          const choices = [...groups, ...nonGroups]
+          return {
+            key: 'system',
+            valueType: 'string',
+            choices: choices,
+            defaultValue: InputSystemGroup
+          }
+        }
+      },
+      out: { flow: 'flow', ...outputSockets },
+      initialState: initialState(),
+      init: ({ read, write, commit, graph }) => {
+        //read from the read and set dict acccordingly
+
+        const system = read<SystemUUID>('system')
+
+        const queue = defineActionQueue(ActionDefinitions[type].matches)
+        queue() // flush the queue
```

Comment:
Why are we flushing the queue?

Reply:
when we restart the graph, by pausing and playing it retains the queue from the old run, this ensures that we flush the queue before running the behave graph, this is mainly for the studio for rapid testing and iteration and also if we intend to run the graph multiple times in a location